modules/hydrodyn/src/WAMIT.f90

```diff
@@ -902,13 +901,14 @@ SUBROUTINE WAMIT_Init( InitInp, u, p, x, xd, z, OtherState, y, m, Interval, ErrS
       else
            ! Initialize the variables associated with the incident wave:
 
-         SELECT CASE ( InitInp%WaveMod ) ! Which incident wave kinematics model are we using?
-         CASE ( 0 )  ! No waves, NOTE: for this case we are forcing ExctnDisp = 0, so only p%WaveExctn needs to be allocated, not p%WaveExctnGrid
+         SELECT CASE ( InitInp%WaveField%WaveMod ) ! Which incident wave kinematics model are we using?
```

Comment:
Is there a reason that `InitInp%WaveField` isn't `p%WaveField`? It looks like it was changed in most other places.

Reply:
Addressed in 67c77c: it was just an oversight. I think I had added the parameter p%WaveField after switching a few of the variables.